testdata/baselines/reference/submodule/compiler/noInferUnionExcessPropertyCheck1.types

```diff
@@ -2,20 +2,20 @@
 
 === noInferUnionExcessPropertyCheck1.ts ===
 declare function test1<T extends { x: string }>(
->test1 : <T extends { x: string; }>(a: T, b: () => T | T) => void
+>test1 : <T extends { x: string; }>(a: T, b: () => NoInfer<T> | NoInfer<T>) => void

```

Comment:
Nothing you did but the printer definitely mishandles precedence of arrow functions and unions 😄

Reply:
Yep, I noticed that too - the reason is that the current printer doesn't have any support for parenthesizing rules yet (the old one has a concept of `parenthesizerRule`)

Reply:
The current printer is just a placeholder, but IIRC the new compiler doesn't use parenthesizer rules at all anymore.

Reply:
Ye, might be - I was taking another look at this a moment ago and it seems that the node builder “transforms” those into ParenthesizedTypes (and then those are simply printed)